src/OpenTelemetry.AutoInstrumentation/Logging/OtelLogging.cs

```diff
@@ -0,0 +1,140 @@
+﻿// <copyright file="OtelLogging.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using OpenTelemetry.AutoInstrumentation.Configuration;
+
+namespace OpenTelemetry.AutoInstrumentation.Logging;
+
+/// <summary>
+/// Configures shared logger used by instrumentations.
+/// </summary>
+internal static class OtelLogging
+{
+    private const string NixDefaultDirectory = "/var/log/opentelemetry/dotnet";
+
+    private static readonly ILogger Logger;
+
+    static OtelLogging()
+    {
+        ISink sink = null;
+        try
+        {
+            var logDirectory = GetLogDirectory();
+            if (logDirectory != null)
+            {
+                var fileName = GetLogFileName();
+                var logPath = Path.Combine(logDirectory, fileName);
+                sink = new FileSink(logPath);
+            }
+        }
+        catch (Exception)
+        {
+            // unable to configure logging to a file
+        }
+
+        if (sink == null)
+        {
+            sink = new NoopSink();
+        }
+
+        Logger = new Logger(sink);
+    }
+
+    internal static ILogger GetLogger() => Logger;
+
+    private static string GetLogFileName()
+    {
+        try
+        {
+            using var process = Process.GetCurrentProcess();
+            var appDomainName = AppDomain.CurrentDomain.FriendlyName;
+
+            return $"dotnet-tracer-managed-{appDomainName}-{process.Id}.log";
+        }
+        catch
+        {
+            // We can't get the process info
+            return $"dotnet-tracer-managed-{Guid.NewGuid()}.log";
+        }
+    }
+
+    private static string GetLogDirectory()
+    {
+        string logDirectory;
+
+        try
+        {
+            logDirectory = Environment.GetEnvironmentVariable(ConfigurationKeys.LogDirectory);
+
+            if (logDirectory == null)
+            {
+                var nativeLogFile = Environment.GetEnvironmentVariable(ConfigurationKeys.LogPath);
```

Comment:
how about renaming `nativeLogFile` to something more or less like `envVarValue` ?

Reply:
fixed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/456/commits/3bd2baa6343ce26aa9e37e50f7aed7c829d91951